axon-server-connector/src/main/java/org/axonframework/axonserver/connector/util/AxonFrameworkVersionResolver.java

```diff
@@ -0,0 +1,86 @@
+package org.axonframework.axonserver.connector.util;
+
+
+import org.axonframework.util.MavenArtifactVersionResolver;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Provides Axon Framework current version.
+ * <p>
+ * At first it looks for the version in the Meta Data of the maven artifact.
+ * As a fallback it looks for the version in the environment variables.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2.1
+ */
+public class AxonFrameworkVersionResolver implements Supplier<String> {
+
+    private static final String ORG_AXONFRAMEWORK = "org.axonframework";
+    private static final String AXON_SERVER_CONNECTOR = "axon-server-connector";
+    private static final String AXON_FRAMEWORK_VERSION = "AXON_FRAMEWORK_VERSION";
+    private static final String UNKNOWN_VERSION = "";
+
+    private final Supplier<String> mavenAxonVersionSupplier;
+    private final Function<String, String> envPropertySupplier;
+
+    private final AtomicReference<String> version = new AtomicReference<>();
+
+    /**
+     * Creates an instance that uses a {@link MavenArtifactVersionResolver} to resolve the Axon Server Connector jar
+     * version, and {@link System#getenv(String)} to resolve the AXON_SERVER_CONNECTOR environment property as a
```

Comment:
Curious: Is `AXON_SERVER_CONNECTOR` intentionally capitalized?

Reply:
By convention, I'm used defining env properties all uppercase, separated with underscores. We can change it if you prefer. This fallback is useful for development mainly.

Reply:
Ah, I wasn't reading the javadoc properly...please disregard my request.